org.eclipse.xtext.xtext.ui.tests/src/org/eclipse/xtext/xtext/ui/editor/folding/XtextFoldingTest.xtend

```diff
@@ -0,0 +1,242 @@
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.xtext.xtext.ui.editor.folding
+
+import org.eclipse.xtext.testing.InjectWith
+import org.eclipse.xtext.testing.XtextRunner
+import org.eclipse.xtext.ui.testing.AbstractFoldingTest
+import org.eclipse.xtext.xtext.ui.XtextUiInjectorProvider
+import org.junit.Test
+import org.junit.runner.RunWith
+
+/**
+ * @author miklossy - Initial contribution and API
+ */
+@RunWith(XtextRunner)
+@InjectWith(XtextUiInjectorProvider)
+class XtextFoldingTest extends AbstractFoldingTest {
+
+	@Test def greetings() {
+		'''
+			grammar org.xtext.example.mydsl.MyDsl with org.eclipse.xtext.common.Terminals
+			
+			generate myDsl "http://www.xtext.org/example/mydsl/MyDsl"
+			
+			[>Model:
+				greetings+=Greeting*;<]
+				
+			[>Greeting:
+				'Hello' name=ID '!';<]
+		'''.testFoldingRegions
+	}
+
+	@Test def arithmetics() {
+		'''
+			[>/*******************************************************************************
+			 * Copyright (c) 2015, 2017 itemis AG (http://www.itemis.eu) and others.
+			 * This program and the accompanying materials are made available under the
+			 * terms of the Eclipse Public License 2.0 which is available at
+			 * http://www.eclipse.org/legal/epl-2.0.
+			 *
+			 * SPDX-License-Identifier: EPL-2.0
+			 *******************************************************************************/<]
+			grammar org.eclipse.xtext.example.arithmetics.Arithmetics with org.eclipse.xtext.common.Terminals
+			
+			import "http://www.eclipse.org/emf/2002/Ecore" as ecore
+			generate arithmetics "http://www.eclipse.org/Xtext/example/Arithmetics"
+			
+			[>Module:
+				'module' name=ID
+				(imports+=Import)*
+				(statements+=Statement)*;<]
+			
+			[>Import:
+				'import' module=[Module];<]
+			
+			[>Statement:
+				Definition | Evaluation;<]
+			
+			[>Definition:
+				'def' name=ID ('(' args+=DeclaredParameter (',' args+=DeclaredParameter)* ')')?
+				':' expr=Expression ';';<]
+			
+			[>DeclaredParameter:
+				name=ID;<]
+			
+			[>AbstractDefinition:
+				Definition | DeclaredParameter;<]
+			
+			[>Evaluation:
+				expression=Expression ';';<]
+			
+			[>Expression:
+				Addition;<]
+			
+			[>Addition returns Expression:
+				Multiplication (({Plus.left=current} '+' | {Minus.left=current} '-') right=Multiplication)*;<]
+			
+			[>Multiplication returns Expression:
+				PrimaryExpression (({Multi.left=current} '*' | {Div.left=current} '/') right=PrimaryExpression)*;<]
+			
+			[>PrimaryExpression returns Expression:
+				'(' Expression ')' |
+				{NumberLiteral} value=NUMBER |
+				{FunctionCall} func=[AbstractDefinition] ('(' args+=Expression (',' args+=Expression)* ')')?;<]
+			
+			[>terminal NUMBER returns ecore::EBigDecimal:
+				('0'..'9')* ('.' ('0'..'9')+)?;<]
+			
+			[>@Override
+			terminal INT returns ecore::EInt:
+				'this one has been deactivated';<]
+			}
```

Comment:
What is this curly brace here?

Reply:
Good catch eagle-eye @szarnekow ! I removed it.

Reply:
I wonder why the test passed and if it is a good idea to accept syntax errors by default in the folding test utilities.

Reply:
We could improve that in the future, but I think for the time being it is acceptable.

Reply:
i think currently it is a feature to test folding on broken models

Reply:
> i think currently it is a feature to test folding on broken models

Yes, it is, but it could be a good idea to make it configurable in the AbstractFoldingTest base class if testing broken models is desired or not.